sources/corepackUtils.ts

```diff
@@ -109,11 +109,9 @@ export async function installVersion(installTarget: string, locator: Locator, {s
   const {version, build} = locatorReference;
 
   const installFolder = path.join(installTarget, locator.name, version);
-  const corepackFile = path.join(installFolder, `.corepack`);
 
-  // Older versions of Corepack didn't generate the `.corepack` file; in
-  // that case we just download the package manager anew.
-  if (fs.existsSync(corepackFile)) {
+  if (fs.existsSync(installFolder)) {
+    const corepackFile = path.join(installFolder, `.corepack`);
     const corepackContent = await fs.promises.readFile(corepackFile, `utf8`);
     const corepackData = JSON.parse(corepackContent);
```

Comment:
While we're at it, we could probably skip the `existSync` call and catch the `ENOENT` error instead
```suggestion
  try {
    const corepackFile = path.join(installFolder, `.corepack`);
    const corepackContent = await fs.promises.readFile(corepackFile, `utf8`);
    const corepackData = JSON.parse(corepackContent);
```